### PR TITLE
Decompile func_800E38CC

### PR DESCRIPTION
**Pull request by sozud**

### CODE REVIEW

File: include/dra.h [hunk 1]
```diff
@@ -961,6 +964,7 @@ extern s32 D_801362F4;
 extern s32 D_801362F8;
 extern s32 D_801362FC;
 extern s32 g_softResetTimer;
+extern s16 D_80136308;
```
Comment (Xeeynamo) [inline include/dra.h:967]: This is an array of s16: `extern s16 D_80136308[]`

File: src/dra/42398.c [hunk 2]
```diff
@@ -227,7 +227,20 @@ void func_800E376C(void) {
...
-INCLUDE_ASM("asm/dra/nonmatchings/42398", func_800E38CC);
+void func_800E38CC(void) {
+    if (D_800A015C != 0) {
+        if ((u32)D_800A0158 >= 0x24U) {
+            D_801390D4->buf.disp.screen.x = 0;
+            PutDispEnv(&D_801390D4->buf.disp);
+            return;
+        }
+        D_801390D4->buf.disp.screen.x =
+            (s16) * (&D_80136308 + ((D_800A0158 + D_800A04F8)));
```
Comment (Xeeynamo) [inline src/dra/42398.c:238]: whenever you see `(&D_80012345 + something)` that is almost always rewritable as `D_80012345[something]`

File: src/dra/42398.c [hunk 3]
```diff
@@ -227,7 +227,20 @@ void func_800E376C(void) {
...
+        PutDispEnv(&D_801390D4->buf.disp);
+        D_800A0158 += 1;
```
Comment (Xeeynamo) [inline src/dra/42398.c:240]: can get rid of `+=1` for a simpler `++`

File: src/dra/42398.c [hunk 4]
```diff
@@ -227,7 +227,20 @@ void func_800E376C(void) {
 // mips to c doesn't support this function very well
 INCLUDE_ASM("asm/dra/nonmatchings/42398", func_800E385C);
 
-INCLUDE_ASM("asm/dra/nonmatchings/42398", func_800E38CC);
+void func_800E38CC(void) {
+    if (D_800A015C != 0) {
+        if ((u32)D_800A0158 >= 0x24U) {
```
Comment (Xeeynamo) [inline src/dra/42398.c:232]: Whenever possible, try getting rid of redundant casts to make the code leaner

File: src/dra/42398.c [hunk 5]
```diff
@@ -227,7 +227,20 @@ void func_800E376C(void) {
...
+            D_801390D4->buf.disp.screen.x = 0;
+            PutDispEnv(&D_801390D4->buf.disp);
+            return;
```
Comment (Xeeynamo) [inline src/dra/42398.c:235]: You can get rid of this return by using an `else`. I believe that is how developers wrote the real code.

**Review by Xeeynamo — CHANGES_REQUESTED** [review]
Left some comments

**Comment by sozud** [comment]
Thanks, I think the comments should be resolved now.

**Review by Xeeynamo — APPROVED** [review]
All good. Thank you for the contribution!

**Pull request merged**